aep/general/0004/aep.md

```diff
@@ -16,65 +16,99 @@ tools such as Kubernetes or GraphQL interact with APIs from multiple providers.
 APIs **must** define a resource type for each resource in the API, according to
 the following pattern: `{API Name}/{Type Name}`. The type name:
 
-- **must** Start with an uppercase letter.
-- **must** Only contain alphanumeric characters.
+- **must** Only contain ASCII alphanumeric characters.
+- **must** Start with a lowercase letter.
 - **must** Be of the singular form of the noun.
-- **must** Use PascalCase (UpperCamelCase).
-- For Kubernetes, the type name **must** match the [object][] name.
-- For OpenAPI, the type name **must** match the name of the schema representing
-  the object.
-- For Protobuf, the type name **must** match the name of the protobuf message.
+- **must** Use kebab case.
+- For Kubernetes, the type name when converted to upper camel case **must**
+  match the [object][] name.
+- For OpenAPI, the type name when converted to upper camel case **must** match
+  the name of the schema representing the object.
+- For protobuf, the type name when converted to upper camel case **must** match
+  the name of the protobuf message.
 
 ### Examples
 
 Examples of resource types include:
 
-- `pubsub.example.com/Topic`
-- `pubsub.example.com/Subscription`
-- `spanner.example.com/Database`
-- `spanner.example.com/Instance`
-- `networking.istio.io/Instance`
+- `networking.istio.io/instance`
+- `pubsub.example.com/topic`
+- `pubsub.example.com/subscription`
+- `spanner.example.com/database`
+- `spanner.example.com/instance`
+- `user.example.com/user-event`
 
 ### Annotating resource types
 
-APIs **must** annotate the resource types for each resource in the API using
-the [`google.api.resource`][resource] annotation:
+APIs **must** annotate the resource types for each resource in the API
+
+{% tab proto %}
+
+For protobuf, use the [`google.api.resource`][resource] annotation:
 
 ```proto
-// A representation of a Pub/Sub topic.
+// A representation of a user event.
 message Topic {
   option (google.api.resource) = {
-    type: "pubsub.googleapis.com/Topic"
-    pattern: "projects/{project}/topics/{topic}"
-    singular: "topic"
-    plural: "topics"
+    type: "user.example.com/user-event"
+    singular: "user-event"
+    plural: "user-events"
+    // define one or more patterns, e.g. if a resource has more than one parent.
+    pattern: "projects/{project}/user-events/{user-event}"
+    pattern: "folder/{folder}/user-events/{user-event}"
+    pattern: "users/{user}/events/{user-event}"
   };
 
   // Name and other fields...
+```
+
+{% tab oas %}
+
+For OpenAPI 3.0, use the `x-aep-resource` extension:
+
+```json
+{
+  "type": "object",
+  "x-aep-resource": {
```

Comment:
Is this extension defined anywhere e.g. as a JSON Schema?  (Does it need to be?)

Reply:
it's a good question! It definitely should be documented somewhere.

The question is where. I'm thinking maybe one of:

1. an AEP
2. in something like your aep repository


Personally I err toward something a little more fully encapsulated, and think we should document it in this repo. But I'm flexible. Definitely a topic we can discuss as a group?

Reply:
schema added, ptal.

Reply:
I'm not sure I understand the goal here from the snippets of conversation. It might help me to have an explanation of what `x-aep-resource` is meant to capture? I suspect there's a limitation in the OpenAPI that you're trying to compensate for?